### PR TITLE
Add support polykind for `Quote` from `Prim.TypeError`

### PR DESCRIPTION
**Pull request by xgrommx**
Previously `Quote` working only with kind `Type`. Now it will be support any kind

For Example

```purescript
data X :: forall k. k -> Type
data X a

foreign import data MkX :: forall k. k -> X k

e = ("a" :: forall a. Fail (Beside (Text "Some problem with ") (Quote (MkX Number))) => String)

main = log e
```

Will be produce

```purescript
  A custom type error occurred while solving type class constraints:

    Some problem with MkX @Type Number
```

### CODE REVIEW

**Review by JordanMartinez — APPROVED** [review]

File: tests/purs/failing/ProgrammablePolykindedTypeErrorsTypeString.out [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Error found:
+in module [33mMain[0m
+at tests/purs/failing/ProgrammablePolykindedTypeErrorsTypeString.purs:23:7 - 23:17 (line 23, column 7 - line 23, column 17)
+
+  A custom type error occurred while solving type class constraints:
+
+    Don't want to show Just @Type String because.
```
Comment (hdgarrood) [inline tests/purs/failing/ProgrammablePolykindedTypeErrorsTypeString.out:7]: I think we should probably strip the kind annotations out here (the `@Type`) bit.

**Review by hdgarrood — CHANGES_REQUESTED** [review]

File: tests/purs/failing/ProgrammablePolykindedTypeErrorsTypeString.out (hunk 1 above)
Comment (rhendric) [reply tests/purs/failing/ProgrammablePolykindedTypeErrorsTypeString.out:7]: We still want to show them when `--verbose-errors` is set, right?

Comment (hdgarrood) [reply tests/purs/failing/ProgrammablePolykindedTypeErrorsTypeString.out:7]: If we already decided on a strategy of showing kind annotations when verbose errors is on, then applying that same strategy here sounds good. I can’t remember whether that should be happening anyway; do the tests run with verbose errors on?

Comment (rhendric) [reply tests/purs/failing/ProgrammablePolykindedTypeErrorsTypeString.out:7]: As far as I can tell, in the only places we're currently erasing kind annotations in error messages, we do so conditional on verbose errors being off. I believe the tests don't run with verbose errors, which maybe makes testing that the erasure is conditional on verbosity difficult.

Also, it seems there are plenty of other places where we don't do this and perhaps should (just grep for `@` in `tests/**/*.out`), which maybe makes dumping this request on this PR a little unfair? I agree that it should be done, but maybe we should clean this up holistically (and figure out how to test it robustly) in a separate PR.

Comment (hdgarrood) [reply tests/purs/failing/ProgrammablePolykindedTypeErrorsTypeString.out:7]: Ok, yes, that makes sense.

**Review by hdgarrood — APPROVED** [review]

File: tests/purs/failing/ProgrammablePolykindedTypeErrorsTypeString.out (hunk 1 above)
Comment (natefaubion) [reply tests/purs/failing/ProgrammablePolykindedTypeErrorsTypeString.out:7]: We don’t strip them in the error “trace” to aid debugging.

**Comment by rhendric** [comment]
@xgrommx, could you please:

1. Rebase this on top of current master.
2. Create a file in CHANGELOG.d for this change—it doesn't have to be more than a brief bullet point. (See CHANGELOG.d/README.md, after you've rebased.)

When that's squared away, we can merge!

**Comment by xgrommx** [comment]
@rhendric what do u mean about rebase?

**Comment by JordanMartinez** [comment]
@xgrommx he means `git rebase`. We've changed how we handle the changelog (see #4132), so the rebase needs to occur so that you can add a file in that directory.
```
git fetch upstream
git rebase upstream/master fix-type-error-quote
# add a file to CHANGELOG.d (see its README.md)
# to document this change
```

**Comment by JordanMartinez** [comment]
@xgrommx Any updates on this?

**Comment by xgrommx** [comment]
@JordanMartinez sorry I was in vacation.

**Comment by xgrommx** [comment]
It's ok?

```
purescript-2:fix-type-error-quote* λ git fetch origin
purescript-2:fix-type-error-quote* λ git rebase origin/master fix-type-error-quote
Current branch fix-type-error-quote is up to date.
purescript-2:fix-type-error-quote* λ
```

**Comment by JordanMartinez** [comment]
@xgrommx Ah, no worries about that.
 
Your example above is fetching data from `origin` (i.e. your fork of purescript), not `upstream` (i.e the actual purescript repo). Similarly, it's rebasing against your fork's `master` branch, not the purescript repo's `master` branch.

Below is what I get when I run `git remote -v`, What do you get when you run that?
```bash
$ git remote -v
origin	git@github.com:JordanMartinez/purescript.git (fetch)
origin	git@github.com:JordanMartinez/purescript.git (push)
rh	git@github.com:rhendric/purescript.git (fetch)
rh	git@github.com:rhendric/purescript.git (push)
upstream	https://github.com/purescript/purescript.git (fetch)
upstream	https://github.com/purescript/purescript.git (push)
```

**Comment by JordanMartinez** [comment]
Rather than rebasing on `master`, I merged with master and added a changelog entry. I haven't tested this out, but I assume CI should still pass.

**Pull request merged**